addons/sale_project/models/sale_order.py

```diff
@@ -163,7 +163,7 @@ def action_view_task(self):
             action['res_id'] = self.tasks_ids.id
         # set default project
         default_line = next((sol for sol in self.order_line if sol.product_id.detailed_type == 'service'), self.env['sale.order.line'])
-        default_project_id = default_line.project_id.id or self.project_id.id or self.project_ids[:1].id
+        default_project_id = default_line.project_id.id or self.project_id.id or self.project_ids[:1].id or self.tasks_ids.project_id[-1:].id
```

Comment:
Why did we choose to select the last project instead of the first one as we did for `self.project_ids`? 🧐

Reply:
Sorry my bad I got confused in this. You are right it should be the first one. :sweat_smile: